### PR TITLE
Implement the _catalog endpoint

### PR DESCRIPTION
**Pull request by lubosmj**
closes #5772
https://pulp.plan.io/issues/5772

### CODE REVIEW

**Comment by lubosmj** [comment]
This PR contains changes introduced in the PR https://github.com/pulp/pulp_container/pull/38 to let the tests pass without failures.

File: CHANGES/5772.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Let users fetch the list of all repositories via the _catalog endpoint
```
Comment (ipanova) [inline CHANGES/5772.feature]: s/ all repositories/ all distributed repositories

**Comment by ipanova** [comment]
@lubosmj i suggest adding a new section in the docs about the new endpoint https://pulp-container.readthedocs.io/en/latest/workflows/index.html + add a script to docs/_scripts wdyt?

**Comment by lubosmj** [comment]
> @lubosmj i suggest adding a new section in the docs about the new endpoint https://pulp-container.readthedocs.io/en/latest/workflows/index.html + add a script to docs/_scripts wdyt?

I am afraid I cannot just simply add a new script to docs/_scripts because we need to set up the token authentication first. For now, I have only created a separate section in the docs describing the basic usage of the _catalog endpoint. I think this could satisfy our requirements well enough.

**Pull request merged**